### DISCUSSION TITLE
Some issues on Android

### DISCUSSION THREAD

**Discussion by aaadipop** [HaylLtd/cordova-plugin-background-geolocation#172] [General]
Hello,

Great plugin :) thanks,

I have a question,

On IOS everithing is super smooth and works as expected but on Android everything seems a bit broked and idk why

my config is located in an [Angular service](https://gist.github.com/aaadipop/8ecea5a9977d5f5cf24ff696fc5a892f) and when I hit the `start()` button on some phones takes a while and when I put the app in background the location isn't sent to the server, this is the biggest issue

some conslusions:
1. the location tracking isn't close on terminate
2. the `BackgroundGeolocation.configure` don't apply configs on the flight and idk exactly if I need to wait to stop() or not
3. I can't debug the app, I tried the example from docs but I got smtg like `unexpected token 1 at JSON position 1`
4. the timeout don't work so well after calling config() second time, I'm trying to get the best possible location as fast as possible and then send it 4 times / minute ( or even less ), but this one I managed with a one `getCurrentLocation()` call after init
5. if I'm trying to send location in `BackgroundGeolocation.on("location")` callback with a self httpPost works for a short period of time and after all locations are stored locally and sent on resume
7. no other errors in console

what I'm missing?

thanks,

**Comment by HarelM** [comment]
Cordova/capacitor will stop the webview event loop when in the background, you need the http configuration.
I can debug the plugin in my, so you'll need to continue fiddling.
I can't comment about the rest, sorry, here is how I use it:
https://github.com/IsraelHikingMap/Site/blob/ed8ed0f367b7479db1165ab13176bc0962d8a077/IsraelHiking.Web/src/application/services/geo-location.service.ts

**Comment by aaadipop** [comment]
thanks for the fast reply, I'll try your example

**Comment by aaadipop** [comment]
yep, with http configuration is the issue, but I'll try again

**Comment by aaadipop** [comment]
all good, thanks
